cmake error on a clean build

Thx a lot for sharing the repo! I have been trying to find a repo template for cpp with support for CI and your repo is a great template for my project.

However, I was unable to run `cmake .. -DCMAKE_BUILD_TYPE=coverage` after following the steps. 

The error is as follow 
```
-- The CXX compiler identification is GNU 7.5.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done-- Found Doxygen: /usr/bin/doxygen (found version "1.8.13") found components:  doxygen dot 
-- IPO is supported
CMake Error at CMakeLists.txt:58 (target_link_libraries):
  Object library target "example" may not link to anything.


-- Configuring incomplete, errors occurred!
See also "/workspaces/PoissonDiscSampling/build/CMakeFiles/CMakeOutput.log".
root@c067bbd707d0:/workspaces/PoissonDiscSampling/build# mkdir build && cd build && cmake .. -DCMAKE_BUILD_TYPE=coverage^C
root@c067bbd707d0:/workspaces/PoissonDiscSampling/build# cd ../^C
root@c067bbd707d0:/workspaces/PoissonDiscSampling/build# rm -rf ./*
root@c067bbd707d0:/workspaces/PoissonDiscSampling/build# mkdir build && cd build && cmake .. -DCMAKE_BUILD_TYPE=coverage -DENABLE_DOCTESTS=OFF
CMake Error: The source directory "/workspaces/PoissonDiscSampling/build" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
root@c067bbd707d0:/workspaces/PoissonDiscSampling/build/build# cd ..
root@c067bbd707d0:/workspaces/PoissonDiscSampling/build# rm -rf ./*
root@c067bbd707d0:/workspaces/PoissonDiscSampling/build# cmake .. -DCMAKE_BUILD_TYPE=coverage -DENABLE_DOCTESTS=OFF
-- The CXX compiler identification is GNU 7.5.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.13") found components:  doxygen dot 
-- IPO is supported
CMake Error at CMakeLists.txt:58 (target_link_libraries):
  Object library target "engine" may not link to anything.


-- Configuring incomplete, errors occurred!
```
Thought it might be related to the git submodule status, I try using `git submodule add ` to manually add doctest but still no luck.

__code to reproduce__ (cmake version 3.10.2):
```
git clone https://github.com/bsamseth/cpp-project.git
cd cpp-project
sh setup.sh
mkdir build && cd build
cmake .. -DCMAKE_BUILD_TYPE=coverage
```

Github Action would also return failure during build step, see [here](https://github.com/wyq977/cpp-example/runs/526486319?check_suite_focus=true)

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -9,13 +9,8 @@ jobs:
     
     steps:
     - uses: actions/checkout@v2
-    - name: Checkout submodules
-      shell: bash
-      run: |
-        auth_header="$(git config --local --get http.https://github.com/.extraheader)"
-        git submodule sync --recursive
-        git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
-        
+      with:
+          submodules: true
     - name: configure
       run: mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE=debug ..
     - name: build
@@ -30,13 +25,8 @@ jobs:
     
     steps:
     - uses: actions/checkout@v2
-    - name: Checkout submodules
-      shell: bash
-      run: |
-        auth_header="$(git config --local --get http.https://github.com/.extraheader)"
-        git submodule sync --recursive
-        git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
-        
+      with:
+          submodules: true
     - name: configure
       run: |
         mkdir build

--- a/README.md
+++ b/README.md
@@ -88,8 +88,8 @@ upload this to Coveralls.io and/or Codecov.io
 When starting a new project, you probably don't want the history of this repository. To start fresh you can use
 the [setup script](setup.sh) as follows:
 ``` bash
-> git clone https://github.com/bsamseth/cpp-project  # Or use ssh-link if you like.
+> git clone --recurse-submodules https://github.com/bsamseth/cpp-project  # Or use ssh-link if you like.
 > cd cpp-project
-> sh setup.sh
+> bash setup.sh
 ```
 The result is a fresh Git repository with one commit adding all files from the boiler plate. 

--- a/setup.sh
+++ b/setup.sh
@@ -1,12 +1,5 @@
 #!/bin/bash
 
-# Initialize submodules: This should already be done when cloning, but there are ways to muck it
-# up if you do things in the wrong order. So just to be sure, we do it now.
-git submodule update --init --recursive
-
-# Remove the remote (you probably want your own instead).
-git remote remove origin
-
 # Revert to first commit, add and commit everything as single commit.
 git reset "$(git rev-list --max-parents=0 --abbrev-commit HEAD)"
 
@@ -23,3 +16,9 @@ else
 	git commit --amend --author="$name <$email>" 
 fi
 
+# Initialize submodules: This should already be done when cloning, but there are ways to muck it
+# up if you do things in the wrong order. So just to be sure, we do it now.
+git submodule update --init --recursive
+
+# Remove the remote (you probably want your own instead).
+git remote remove origin

--- a/setup.sh
+++ b/setup.sh
@@ -1,4 +1,8 @@
-#!/bin/sh
+#!/bin/bash
+
+# Initialize submodules: This should already be done when cloning, but there are ways to muck it
+# up if you do things in the wrong order. So just to be sure, we do it now.
+git submodule update --init --recursive
 
 # Remove the remote (you probably want your own instead).
 git remote remove origin
@@ -19,8 +23,3 @@ else
 	git commit --amend --author="$name <$email>" 
 fi
 
-# Initialize submodules: This is done by CMake, but there are ways to muck it
-# up if you does things in the wrong order. So just to be sure, we do it now.
-git submodule init
-
-git submodule update

--- a/setup.sh
+++ b/setup.sh
@@ -1,9 +1,7 @@
 #!/bin/bash
 
 # Revert to first commit, add and commit everything as single commit.
-git reset "$(git rev-list --max-parents=0 --abbrev-commit HEAD)"
-
-git add --all
+git reset $(git commit-tree HEAD^{tree} -m "Bolier plate for C++ project added")
 
 name=$(git config user.name)
 email=$(git config user.email)

